optimum/habana/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl_mlperf.py

```diff
@@ -897,18 +889,20 @@ def __call__(
 
             output_images.append(image)
 
+
         hb_profiler.stop()
 
         speed_metrics_prefix = "generation"
         ht.hpu.synchronize()
-
+        if use_warmup_inference_steps:
+            ht.hpu.synchronize()
+            t1 = warmup_inference_steps_time_adjustment(t1, t1, num_inference_steps, throughput_warmup_steps)
         if t1 == t0 or use_warmup_inference_steps:
-            num_samples = num_batches * batch_size
-            num_steps = (num_inference_steps - throughput_warmup_steps) * num_batches * batch_size
+            num_samples = batch_size
```

Comment:
There could still be more than 1 batch no?

Reply:
we just calculate the latest batch time in use_warmup_inference_steps case(add and j == num_batches - 1 in line 841), so the  num_samples is batch_size